Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,144 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `type`
+    public enum IncidentType: String {
```

Comment:
The full name of this type would be `Incident.IncidentType`, which is redundant. Swift already takes up the `Type` name, so we could call it `Kind`, for consistency with `VisualInstruction.Component.Kind`.

Reply:
'Kind' sound a bit odd to my ear, but if there are already such cases I assume that is reasonable to use.